Compiling with -std=c17 results in platform not supported error

```
/home/m/static-php-cli/source/php-src/ext/spx/src/spx_resource_stats.c:28:5: error: #error "Your platform is not supported. Please open an issue."
   28 | #   error "Your platform is not supported. Please open an issue."
      |     ^~~~~
cc1: note: unrecognized command-line option ‘-Wno-typedef-redefinition’ may have been intended to silence earlier diagnostics
make: *** [Makefile:240: src/spx_resource_stats.lo] Error 1
make: *** Waiting for unfinished jobs....
In file included from /home/m/static-php-cli/source/php-src/ext/spx/src/php_spx.c:29:
/home/m/static-php-cli/source/php-src/ext/spx/src/php_spx.h:27:5: error: #error "Only Linux-based OSes, Apple MacOS and FreeBSD are supported"
   27 | #   error "Only Linux-based OSes, Apple MacOS and FreeBSD are supported"
      |     ^~~~~
cc1: note: unrecognized command-line option ‘-Wno-typedef-redefinition’ may have been intended to silence earlier diagnostics
make: *** [Makefile:216: src/php_spx.lo] Error 1
```

With `-std=gnu17` or leaving it empty, it compiles. x86_64 and aaarch64 linux gnu.

--- a/config.m4
+++ b/config.m4
@@ -19,7 +19,7 @@ if test "$PHP_SPX" = "yes"; then
     AC_DEFINE_UNQUOTED([SPX_HTTP_UI_ASSETS_DIR], [ "$PHP_SPX_ASSETS_DIR/web-ui" ], [path of web-ui assets directory])
     PHP_SUBST([PHP_SPX_ASSETS_DIR])
 
-    CFLAGS="$CFLAGS -Werror -Wall -Wno-attributes -O3 -pthread -std=gnu90"
+    CFLAGS="$CFLAGS -Werror -Wall -Wno-attributes -O3 -pthread -std=c11"
 
     # Disabling typedef-redefinition is required for:
     #   - macOS, see https://github.com/NoiseByNorthwest/php-spx/pull/270

--- a/config.m4
+++ b/config.m4
@@ -19,7 +19,15 @@ if test "$PHP_SPX" = "yes"; then
     AC_DEFINE_UNQUOTED([SPX_HTTP_UI_ASSETS_DIR], [ "$PHP_SPX_ASSETS_DIR/web-ui" ], [path of web-ui assets directory])
     PHP_SUBST([PHP_SPX_ASSETS_DIR])
 
-    CFLAGS="$CFLAGS -Werror -Wall -Wno-attributes -O3 -pthread -std=c11"
+    CFLAGS="$CFLAGS -Werror -Wall -Wno-attributes -O3 -pthread"
+
+    php_ver_num=$(echo $PHP_VERSION | awk -F. '{printf("%d%02d%02d",$1,$2,$3)}')
+    if test "$php_ver_num" -ge 80200; then
+        CFLAGS="$CFLAGS -std=c11"
+        AC_MSG_NOTICE([Adding -std=c11 because PHP >= 8.2])
+    else
+        AC_MSG_NOTICE([Not adding -std=c11 because PHP < 8.2])
+    fi
 
     # Disabling typedef-redefinition is required for:
     #   - macOS, see https://github.com/NoiseByNorthwest/php-spx/pull/270

--- a/config.m4
+++ b/config.m4
@@ -26,7 +26,8 @@ if test "$PHP_SPX" = "yes"; then
         CFLAGS="$CFLAGS -std=c11"
         AC_MSG_NOTICE([Adding -std=c11 because PHP >= 8.2])
     else
-        AC_MSG_NOTICE([Not adding -std=c11 because PHP < 8.2])
+        CFLAGS="$CFLAGS -std=gnu90"
+        AC_MSG_NOTICE([Adding -std=gnu90 because PHP < 8.2])
     fi
 
     # Disabling typedef-redefinition is required for:

--- a/src/php_spx.h
+++ b/src/php_spx.h
@@ -23,7 +23,7 @@
 #include "main/php.h"
 
 /* linux 2.6+ or OSX */
-#if !defined(linux) && !(defined(__APPLE__) && defined(__MACH__)) && !defined(__FreeBSD__)
+#if !defined(__linux__) && !(defined(__APPLE__) && defined(__MACH__)) && !defined(__FreeBSD__)
 #   error "Only Linux-based OSes, Apple MacOS and FreeBSD are supported"
 #endif
 

--- a/src/spx_resource_stats.c
+++ b/src/spx_resource_stats.c
@@ -16,7 +16,7 @@
  */
 
 
-#if defined(linux)
+#if defined(__linux__)
 #   include "spx_resource_stats-linux.c"
 #elif defined(__APPLE__) && defined(__MACH__)
 #   include "spx_resource_stats-macos.c"

